specs/eventing/control-plane.md

```diff
@@ -0,0 +1,1120 @@
+        success or failure is not yet determined.
+    1.  `True` SHOULD indicate that the application is fully reconciled and
+        operating correctly.
+
+    `Unknown` and `True` are specified as SHOULD rather than MUST requirements
+    because there may be errors which prevent functioning which cannot be
+    determined by the API stack (e.g. DNS record configuration in certain
+    environments). Implementations are expected to treat these as "MUST" for
+    factors within the control of the implementation.
+
+1.  For non-`Ready` conditions, any conditions with `severity=""` (aka "Error
+    conditions") must be aggregated into the "Ready" condition as follows:
+
+    1.  If the condition is `False`, `Ready` MUST be `False`.
+    1.  If the condition is `Unknown`, `Ready` MUST be `False` or `Unknown`.
+    1.  If the condition is `True`, `Ready` may be any of `True`, `False`, or
+        `Unknown`.
+
+    Implementations MAY choose to report that `Ready` is `False` or `Unknown`
+    even if all Error conditions report a status of `True` (i.e. there may be
+    additional hidden implementation conditions which feed into the `Ready`
+    condition which are not reported.)
+
+1.  Non-`Ready` conditions with non-error severity MAY be surfaced by the
+    implementation. Examples of `Warning` or `Info` conditions could include:
+    missing health check definitions, scale-to-zero status, or non-fatal
+    capacity limits.
+
+Conditions type names should be chosen to describe positive conditions where
+`True` means that the condition has been satisfied. Some conditions may be
+transient (for example, `ResourcesAllocated` might change between `True` and
+`False` as an application scales to and from zero). It is RECOMMENDED that
+transient conditions be indicated with a `severity="Info"`.
+
+## Resource Lifecycle
+
+### Broker Lifecycle
+
+A Broker represents an Addressable endpoint (i.e. it has a `status.address.url`
+field) which can receive, store, and forward events to multiple recipients based
+on a set of attribute filters (Triggers). Triggers are associated with a Broker
+based on the `spec.broker` field on the Trigger; it is expected that the
+controller for a Broker will also control the associated Triggers. When the
+Broker's `Ready` condition is `true`, the Broker MUST provide a
+`status.address.url` which accepts all valid CloudEvents and MUST forward the
+received events for filtering to each associated Trigger whose `Ready` condition
+is `true`. As described in the [Trigger Lifecycle](#trigger-lifecycle) section,
+a Broker MAY forward events to an associated Trigger which which does not
+currently have a `true` `Ready` condition, including events received by the
+Broker before the Trigger was created.
+
+The annotation `eventing.knative.dev/broker.class` may be used to select a
+particular implementation of a Broker. When a Broker is created, its
+implementation and the `spec.config` field MUST be populated (`spec.config` MAY
```

Comment:
what do you mean by "it implementation .... MUST be populate" ?

Reply:
or is "implementation" a field? if so, let's back-tic it so it's clear it's not an English word... like I assumed :-)

Reply:
The `eventing.knative.dev/broker.class` annotation, I'll clarify.